src/MyEventEmitter.js

```diff
@@ -1,12 +1,93 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  addEvent(name, callback, repeat, prepend) {
+    if (!Object.hasOwnProperty.call(this.events, name)) {
+      this.events[name] = [
+        {
+          repeat,
+          callback,
+        },
+      ];
+
+      return;
+    }
+
+    if (prepend) {
+      this.events[name].unshift({
+        repeat,
+        callback,
+      });
+    } else {
+      this.events[name].push({
+        repeat,
+        callback,
+      });
+    }
+  }
+
+  on(name, callback) {
+    this.addEvent(name, callback, true, false);
+  }
+
+  once(name, callback) {
+    this.addEvent(name, callback, false, false);
+  }
+
+  prependListener(name, callback) {
+    this.addEvent(name, callback, true, true);
+  }
+
+  prependOnceListener(name, callback) {
+    this.addEvent(name, callback, false, true);
+  }
+
+  off(name, listener) {
+    if (!Object.hasOwnProperty.call(this.events, name)) {
+      return;
+    }
+
+    const callbackIndex = this.events[name].findIndex(
+      ({ callback }) => callback === listener
+    );
+
+    if (callbackIndex !== -1) {
+      if (this.events[name].length === 1) {
+        delete this.events[name];
+      } else {
+        this.events[name].splice(callbackIndex, 1);
+      }
+    }
```

Comment:
Avoid nesting if, because it can be flattened like this
```suggestion
    if (callbackIndex === -1) return
    
    if (this.events[name].length === 1) {
      return delete this.events[name];
    }
      
    this.events[name].splice(callbackIndex, 1);
   
```